File: app/src/main/java/com/geridea/trentastico/gui/activities/NoMenuSettings.java
package com.geridea.trentastico.gui.activities;


/*
 * Created with ♥ by Slava on 22/04/2017.
 */

import android.view.MenuItem;

public final class NoMenuSettings implements IMenuSettings {

    public static final NoMenuSettings INSTANCE = new NoMenuSettings();

    private NoMenuSettings() { }

    public static NoMenuSettings getInstance() {
        return INSTANCE;
    }

    @Override
    public int[] getIdsOfMenuItemsToMakeVisible() {
        return new int[0];
    }

    @Override
    public void bindMenuItem(MenuItem item) {
        //Nothing to bind
    }
}

File: app/src/main/java/com/geridea/trentastico/gui/activities/NoMenuSettings.kt
package com.geridea.trentastico.gui.activities


/*
 * Created with ♥ by Slava on 22/04/2017.
 */

import android.view.MenuItem

class NoMenuSettings private constructor() : IMenuSettings {

    override val idsOfMenuItemsToMakeVisible: IntArray
        get() = IntArray(0)

    override fun bindMenuItem(item: MenuItem) = //Nothing to bind
            Unit

    companion object {

        val instance = NoMenuSettings()
    }
}
